aiida/engine/processes/ports.py

```diff
@@ -195,7 +197,7 @@ def validate_port_name(port_name):
         if any([len(entry) > PORT_NAME_MAX_CONSECUTIVE_UNDERSCORES for entry in consecutive_underscores]):
             raise ValueError(f'invalid port name `{port_name}`: more than two consecutive underscores')
 
-    def serialize(self, mapping, breadcrumbs=()):
+    def serialize(self, mapping: Optional[Dict[str, Any]], breadcrumbs: Sequence[str] = ()) -> Optional[Dict[str, Any]]:
```

Comment:
Shouldn't we be more strict here?
```suggestion
    def serialize(self, mapping: Optional[Dict[str, Port]], breadcrumbs: Sequence[str] = ()) -> Optional[Dict[str, Port]]:
```

Reply:
this isn't correct though, the values are not ports, they are data nodes, mappings, or well any value

Reply:
of course, I misread the code there. Please ignore